src/Chain/ToolBox/ToolBox.php

```diff
@@ -50,7 +50,17 @@ public function execute(ToolCall $toolCall): string
         foreach ($this->tools as $tool) {
             foreach ($this->toolAnalyzer->getMetadata($tool::class) as $metadata) {
                 if ($metadata->name === $toolCall->name) {
-                    return $tool->{$metadata->method}(...$toolCall->arguments);
+                    $result = $tool->{$metadata->method}(...$toolCall->arguments);
+
+                    if ($result instanceof \JsonSerializable || is_array($result)) {
+                        return json_encode($result, flags: JSON_THROW_ON_ERROR);
+                    }
+
+                    if (is_integer($result) || is_float($result)) {
```

Comment:
We could add `Stringable` here easily :)

Reply:
Done